File: illustration-of-algorithm/06/main.cpp
#include <iostream>
#include <vector>
#include <stack>
using namespace std;
struct ListNode
{
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};

vector<int> reversePrint(ListNode *head)
{
    stack<int> s;
    ListNode* p = head;
    while(p->next!=NULL)
    {
        s.push(p->val);
        p = p->next;
    }
    s.push(p->val);
    vector<int> r;
    while (!s.empty())
    {
        r.push_back(s.top());
        s.pop();
        cout << r[r.size() -1]<<endl;
    }
    return r;
}

int main()
{
    ListNode a(1),b(3),c(2);
    a.next=&b;b.next=&c;
    auto r = reversePrint(&a);

}

File: illustration-of-algorithm/06/main.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def reversePrint(self, head:ListNode):
        """
        :type head: ListNode
        :rtype: List[int]
        """
        stack = []
        while head:
            stack.append(head.val)
            head = head.next
        return stack[::-1]

a = ListNode(1)
b = ListNode(2)
c = ListNode(3)
a.next = b
b.next = c

s = Solution()
l = s.reversePrint(a)
print(l)